### PR TITLE
Add smooth activation functions

### LINKED ISSUE #347: Implement Classification with Softmax activation function/Logistic in tensorflow and pytorch
I would like to have to have softmax activation function as my variables each probability of the output of the neural network. 
This will help very much, instead of just regression.

### PR DESCRIPTION
**Pull request by pobonomo**
This is in particular to play with nonlinear barrier algorithm. Smooth activations

- sigmoid
- softplus

### CODE REVIEW

File: src/gurobi_ml/keras/keras.py [hunk 1]
```diff
@@ -90,6 +91,9 @@ def __init__(self, gp_model, predictor, input_vars, output_vars=None, **kwargs):
                     )
                 if step.max_value is not None and step.max_value < float("inf"):
                     raise NoModel(predictor, "Only handle ReLU layers without maxvalue")
+            elif isinstance(step, keras.layers.Softmax):
+                # Softmax activation layer (standalone)
+                pass
```
[the remark this answers is not in the document]
Comment (pobonomo) [reply src/gurobi_ml/keras/keras.py]: @copilot this is indeed a mistake and it shouldn't be supported
